internal/agent/device/device.go

```diff
@@ -189,10 +189,48 @@ func (a *Agent) sync(ctx context.Context, current, desired *v1alpha1.RenderedDev
 		return fmt.Errorf("sync device: %w", err)
 	}
 
+	if a.lifecycleManager.StartedDecommissioning() {
```

Comment:
Can we incorporate this into lifecycleManager.AfterUpdate()? I think in your case we can add statusManager as a dependency of  lifecycleManager.

then in that method you can handle all of these status calls.